src/main/java/org/embulk/input/zendesk/services/ZendeskService.java

```diff
@@ -0,0 +1,14 @@
+package org.embulk.input.zendesk.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.embulk.config.TaskReport;
+import org.embulk.input.zendesk.RecordImporter;
+
+public interface ZendeskService
+{
+    boolean isSupportIncremental();
+
+    TaskReport execute(int taskIndex, RecordImporter recordImporter);
```

Comment:
I'd suggest we find the better method name here, in my understanding, `execute` name means we pass the callable function, then executor will execute it, similar as `retry executor`. Our `execute` here will add record into `recordImporter`, should we change to something like `addToRecordImporter` or another name which is less confusing than execute.

Reply:
fixed